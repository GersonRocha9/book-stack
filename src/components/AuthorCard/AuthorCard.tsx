import React from "react";
import Ionicons from "@expo/vector-icons/Ionicons";

import {
  BookImage,
  BookPublicationDate,
  BookTitle,
  Button,
  ButtonsContainer,
  Container,
  InfosContainer,
} from "./styles";

interface AuthorListProps {
  id: number;
  name: string;
  birthdate: string;
  profile_photo: string;
}

const AuthorCard = ({ name, birthdate, profile_photo }: AuthorListProps) => {
  return (
    <Container>
      <BookImage
        source={{
          uri: profile_photo,
        }}
      />
      <InfosContainer>
        <BookTitle>{name}</BookTitle>

        <BookPublicationDate>Data de nascimento: {birthdate}</BookPublicationDate>
      </InfosContainer>

      <ButtonsContainer>
        <Button
          onPress={() => {
            console.log("Edit");
          }}
          title="Edit"
        >
          <Ionicons name="create" size={24} color="black" />
        </Button>

        <Button
          onPress={() => {
            console.log("Delete");
          }}
          title="Delete"
        >
          <Ionicons name="trash" size={24} color="red" />
        </Button>
      </ButtonsContainer>
    </Container>
  );
};

export default AuthorCard;
