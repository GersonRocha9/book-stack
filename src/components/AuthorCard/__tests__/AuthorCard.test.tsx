import React from "react";
import renderer from "react-test-renderer";
import AuthorCard from "../AuthorCard";

// Teste que verifica se o componente AuthorCard estÃ¡ sendo renderizado corretamente
it("renders correctly", () => {
  const tree = renderer
    .create(
      <AuthorCard
        name="J. R. R. Tolkien"
        birthdate="1892"
        profile_photo="https://upload.wikimedia.org/wikipedia/commons/thumb/5/5a/J._R._R._Tolkien.jpg/220px-J._R._R._Tolkien.jpg"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
}, 10000);
