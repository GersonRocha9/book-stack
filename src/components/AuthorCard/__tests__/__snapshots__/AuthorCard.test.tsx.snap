// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomColor": "#eee",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "maxWidth": 400,
        "paddingBottom": 20,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 20,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "https://upload.wikimedia.org/wikipedia/commons/thumb/5/5a/J._R._R._Tolkien.jpg/220px-J._R._R._Tolkien.jpg",
      }
    }
    style={
      Array [
        Object {
          "height": 120,
          "width": 80,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "marginLeft": 20,
          "maxWidth": 200,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#333",
            "fontSize": 18,
            "fontWeight": "bold",
          },
        ]
      }
    >
      J. R. R. Tolkien
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#333",
            "fontSize": 14,
          },
        ]
      }
    >
      Data de nascimento: 
      1892
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "space-between",
          "width": 100,
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 20,
          "height": 40,
          "justifyContent": "center",
          "marginBottom": 10,
          "opacity": 1,
          "width": 40,
        }
      }
    >
      <Text />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 20,
          "height": 40,
          "justifyContent": "center",
          "marginBottom": 10,
          "opacity": 1,
          "width": 40,
        }
      }
    >
      <Text />
    </View>
  </View>
</View>
`;
